### PR TITLE
Add JavaOnlyDetector

### LINKED ISSUE #41: Port JavaOnly/KotlinOnly lint checks
These are good general-purpose lint checks for guardrailing usage of the language.

These would have to be in a separate module.

CC @ZacSweers

### PR DESCRIPTION
**Pull request by ShaishavGandhi**
Added a new `recipes` module as a sort of read-only module for lint-checks. These will mostly contain lint checks that are used in Uber widely but aren't as general-purpose for the community. However, they still have interesting concepts and can be used by the wider community by copying them over. 

CC @ZacSweers

Resolves #41

### CODE REVIEW

File: recipes/build.gradle [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2019. Uber Technologies
```
Comment (ZacSweers) [inline recipes/build.gradle:2]: Nit: 2020 these

File: recipes/src/main/java/com/uber/lintchecks/recipes/guardrails/JavaOnly.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.lintchecks.recipes.guardrails
+
+annotation class JavaOnly
```
Comment (ZacSweers) [inline recipes/src/main/java/com/uber/lintchecks/recipes/guardrails/JavaOnly.kt:18]: Doc

Comment (ShaishavGandhi) [reply recipes/src/main/java/com/uber/lintchecks/recipes/guardrails/JavaOnly.kt:23]: e68e577

File: recipes/build.gradle (hunk 1 above)
Comment (ShaishavGandhi) [reply recipes/build.gradle]: e49102b

**Pull request merged**